tapscript/tx.go

```diff
@@ -71,65 +71,99 @@ func virtualTxInPrevOut(root mssmt.Node) *wire.OutPoint {
 	)
 }
 
+// virtualGenesisTxIn computes the single input of a Taproot Asset virtual
+// transaction where the asset is both a genesis asset and part of an asset
+// group. The input MS-SMT commits to the genesis asset with witnesses removed.
+func virtualGenesisTxIn(newAsset *asset.Asset) (*wire.TxIn, mssmt.Tree, error) {
+	inputTree := mssmt.NewCompactedTree(mssmt.NewDefaultStore())
+
+	// TODO(bhandras): thread the context through.
+	ctx := context.TODO()
+
+	// For genesis assets with a group key, we must clear the witness field
+	// before building the input MS-SMT. We use this copy of the genesis
+	// asset as the input asset for the virtual transaction.
+	copyWithoutWitness := newAsset.Copy()
+	for i := range copyWithoutWitness.PrevWitnesses {
+		copyWithoutWitness.PrevWitnesses[i].TxWitness = nil
+	}
+
+	key := asset.ZeroPrevID.Hash()
+	leaf, err := copyWithoutWitness.Leaf()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	_, err = inputTree.Insert(ctx, key, leaf)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	treeRoot, err := inputTree.Root(context.Background())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO(roasbeef): document empty hash usage here
+	prevOut := virtualTxInPrevOut(treeRoot)
+
+	return wire.NewTxIn(prevOut, nil, nil), inputTree, nil
+}
+
 // virtualTxIn computes the single input of a Taproot Asset virtual transaction.
 // The input prevout's hash is the root of a MS-SMT committing to all inputs of
 // a state transition.
 func virtualTxIn(newAsset *asset.Asset, prevAssets commitment.InputSet) (
 	*wire.TxIn, mssmt.Tree, error) {
 
-	// Genesis assets shouldn't have any inputs committed, so they'll have
-	// an empty input tree.
-	isGenesisAsset := newAsset.HasGenesisWitness()
 	inputTree := mssmt.NewCompactedTree(mssmt.NewDefaultStore())
-	if !isGenesisAsset {
```

Comment:
Yes, this point was never reached with a genesis asset before because the VM would have exited early on any genesis assets before this line.

Now, we have `virtualGenesisTxIn` and a genesis asset should still never appear at this point, so the branch is unneeded.